src/main/java/duke/ui/parser/Command.java

```diff
@@ -0,0 +1,84 @@
+package duke.ui.parser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import duke.exception.*;
+
+public abstract class Command {
+    public enum CommandType {
+        EXIT, LIST, MARK, UNMARK, DEADLINE, EVENT, TODO, DELETE
+    }
+
+    protected CommandType commandType;
+    protected Integer MIN_ARGUMENTS = null;
+    protected Integer MAX_ARGUMENTS = null;
```

Comment:
As per Java coding standard, variable names should be in camelCase, as uppercase is reserved for constants.